celery/app/utils.py

```diff
@@ -114,7 +114,8 @@ def broker_url(self):
     def result_backend(self):
         return (
             os.environ.get('CELERY_RESULT_BACKEND') or
-            self.get('CELERY_RESULT_BACKEND')
+            self.get('CELERY_RESULT_BACKEND') or
+            self.get('result_backend')
```

Comment:
on celery 4.x series, shouldn't result_backend be looked first?

Reply:
> on celery 4.x series, shouldn't result_backend be looked first?

yes, you're right, i will change the order. but do you want to retain the support of CELERY_RESULT_BACKEND?

Reply:
that will be dropped in celery 5 :) so...

Reply:
Emmm... Okay, now celery still 4.x, so let's keep it...